src/main/java/org/folio/spring/DefaultFolioExecutionContext.java

```diff
@@ -28,46 +36,20 @@ public DefaultFolioExecutionContext(FolioModuleMetadata folioModuleMetadata, Map
     this.tenantId = retrieveFirstSafe(okapiHeaders.get(TENANT));
     this.okapiUrl = retrieveFirstSafe(okapiHeaders.get(URL));
     this.token = retrieveFirstSafe(okapiHeaders.get(TOKEN));
-    //TODO: retrieve user name correctly from JWT token
-    this.userName = "NO_USER";
+
+    var optionalUserInfo = JWTokenUtils.parseToken(token);
+    if (optionalUserInfo.isPresent()) {
```

Comment:
there is pretty much logic in the constructor not covered by tests. please add them

Reply:
Tests were added